### PR TITLE
chore: update environment with rustup cache, podman compat

### PR DESCRIPTION
**Pull request by hdhoang**
This is based on my set up on Fedora 36. I have verified that `CONTAINER_TOOL=docker` (using fedora package `moby-engine`), and `CONTAINER_TOOL=podman` can spawn a shell, where `make check` finishes.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -119,9 +120,10 @@ define ENVIRONMENT_EXEC
 			--env INSIDE_ENVIRONMENT=true \
 			--network host \
 			--mount type=bind,source=${CURRENT_DIR},target=/git/vectordotdev/vector \
-			--mount type=bind,source=/var/run/docker.sock,target=/var/run/docker.sock \
+			$(if $(findstring docker,$(CONTAINER_TOOL)),--mount type=bind$(COMMA)source=/var/run/docker.sock$(COMMA)target=/var/run/docker.sock,) \
```
Comment (bruceg) [inline Makefile:123]: Could this be done by escaping the comma with a backslash instead?
```suggestion
			$(if $(findstring docker,$(CONTAINER_TOOL)),--mount type=bind\,source=/var/run/docker.sock\,target=/var/run/docker.sock,) \
```

**Comment by spencergilbert** [comment]
Trying to test this locally - I don't think this dev workflow is supported on M1's, believe we're making some assumptions of x86 in the code (unrelated to the changes here)

File: Makefile (hunk 1 above)
Comment (hdhoang) [reply Makefile:123]: i started out that way, and also tried more `\`, but the docs above `EMPTY` https://www.gnu.org/software/make/manual/make.html#Syntax-of-Functions calls out comma as an example substitution

Comment (bruceg) [reply Makefile:123]: Gotcha, fair enough.

**Review by spencergilbert — APPROVED** [review]
This looks fine to me, @bruceg would you mind giving it another look?

**Review by bruceg — APPROVED** [review]

**Pull request merged**